core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -117,173 +124,116 @@ public TenantObject setEnabled(final boolean flag) {
-     * from the JSON object contained in the <em>trusted-ca</em> property.
+     * If the <em>trusted-ca</em> property contains a JSON object as value,
+     * then this method constructs a trust anchor based on the information contained
+     * in the JSON object.
+     * <p>If the <em>trusted-ca</em> property contains a list of JSON objects, then
+     * this method constructs a trust anchor for each JSON object contained in the list.
+     * 
      * <ol>
-     * <li>If the object contains a <em>cert</em> property then its content is
-     * expected to contain the Base64 encoded (binary) DER encoding of the
-     * trusted certificate. The returned trust anchor will contain this certificate.</li>
-     * <li>Otherwise, if the object contains a <em>public-key</em> and a <em>subject-dn</em>
+     * <li>If the object contains a <em>public-key</em> and a <em>subject-dn</em>
      * property, then the public key property is expected to contain the Base64 encoded
      * DER encoding of the trusted certificate's public key. The returned trust anchor
      * will contain this public key.</li>
-     * <li>Otherwise, this method returns {@code null}.</li>
+     * <li>Otherwise, this method returns an empty list.</li>
      * </ol>
      * <p>
-     * Once a (non {@code null}) trust anchor has been created, it will be cached and
+     * Once a (non empty) list of trust anchors has been created, it will be cached and
      * returned on subsequent invocations of this method.
      * 
-     * @return The trust anchor or {@code null} if no trusted certificate authority
+     * @return The trust anchor or an empty list if no trusted certificate authority
      *         has been set.
      * @throws GeneralSecurityException if the value of the <em>trusted-ca</em> property
-     *             cannot be parsed into a trust anchor, e.g. because of unsupported
-     *             key type, malformed certificate or public key encoding etc. 
+     *              cannot be parsed into a trust anchor, e.g. because of unsupported
+     *              key type, malformed certificate or public key encoding etc.
      */
     @JsonIgnore
-    public TrustAnchor getTrustAnchor() throws GeneralSecurityException {
-
-        if (trustAnchor != null) {
-            return trustAnchor;
-        } else {
-            final X509Certificate cert = getTrustedCertificateAuthority();
-            if (cert != null) {
-                trustAnchor = new TrustAnchor(cert, null);
-                return trustAnchor;
-            } else {
-                return getTrustAnchorForPublicKey(
-                        getProperty(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA, JsonObject.class));
+    public List<TrustAnchor> getTrustAnchors() throws GeneralSecurityException {
+        final List<JsonObject> configs = Optional.ofNullable(getTrustedCAs()).orElse(Collections.emptyList());
+        if (trustAnchors == null || trustAnchors.size() != configs.size()) {
```

Comment:
this is based on the fact that you can add duplicate trusted CAs, right?
In any case, the number of trusted CAs is expected to be very low, i.e. single digits. IMHO it would be totally ok (and much more robust), to simply *always* newly create the TrustAnchor instances from the adapterConfigurations ...

Reply:
yes, i thought about this too.